hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelinePlacementPolicy.java

```diff
@@ -292,10 +294,15 @@ DatanodeDetails fallBackPickNodes(
     int nodesToFind = nodesRequired - results.size();
     for (int x = 0; x < nodesToFind; x++) {
       // Pick remaining nodes based on the existence of rack awareness.
-      DatanodeDetails pick = rackAwareness
-          ? chooseNodeFromNetworkTopology(
-              nodeManager.getClusterNetworkTopologyMap(), anchor, exclude)
-          : fallBackPickNodes(healthyNodes, exclude);
+      DatanodeDetails pick;
+      try {
+        pick = rackAwareness
+            ? chooseNodeFromNetworkTopology(
+            nodeManager.getClusterNetworkTopologyMap(), anchor, exclude)
+            : fallBackPickNodes(healthyNodes, exclude);
+      } catch (SCMException e) {
```

Comment:
I would like to log the exception in debug level log here. Then we could know the fallback happened.

Reply:
Debug log makes sense. 
My thought was: fall back will happen every time when there is only one node on the rack. Under all other scenarios, this fallback logic shouldn't take effects. As a known case, I was reluctant to add logs for it regardless of its log level. But I guess it can be valuable if fallback logic takes effect in other cases.

Reply:
Could we avoid the exception block here, especially if we think the exception block will be the "normal code path" if the rack only has one node?

Eg, if we change `chooseNodeFromNetworkTopology()` to return null or the picked node, and then refactor this code block to look like this:

```
      DatanodeDetails pick = null;
      if (rackAwareness) {
        pick = chooseNodeFromNetworkTopology(
            nodeManager.getClusterNetworkTopologyMap(), anchor, exclude);
      }
      if (pick == null) {
        pick = fallBackPickNodes(healthyNodes, exclude);
        if (rackAwareness) {
          LOG.debug("Failed to choose node based on topology. Fallback " +
              "picks node as: {}", pick);
        }
      }
```

I think that is easier to read and avoid the expensive exception handler. What do you think?

Reply:
This is clean. I will make updates. Thanks for the suggestion.